core/runtime/v2/shim/util_unix.go

```diff
@@ -76,6 +84,47 @@ func SocketAddress(ctx context.Context, socketPath, id string) (string, error) {
 
 // AnonDialer returns a dialer for a socket
 func AnonDialer(address string, timeout time.Duration) (net.Conn, error) {
+	if strings.HasPrefix(address, themeVsock) {
```

Comment:
```suggestion
	proto, addr, ok := strings.Cut(address, ":://")
	if !ok {
		return net.DialTimeout("unix", socket(address).path(), timeout)
	}

	switch proto {
	case themeVsock:
		return dialVosck(addr)
	case themeHybridVsock:
		return dialHybridVsock(addr)
	default:
		return net.DialTimeout("unix", socket(address).path(), timeout)
	}
```

Reply:
good idea, and please take a look at the code change again.